lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1606,8 +1642,13 @@ spv_ostream &operator<<(spv_ostream &O, SPIRVModule &M) {
   O << MI.MemberNameVec << MI.DecGroupVec << MI.DecorateSet << MI.GroupDecVec
     << MI.ForwardPointerVec
     << TopologicalSort(MI.TypeVec, MI.ConstVec, MI.VariableVec,
-                       MI.ForwardPointerVec)
-    << SPIRVNL() << MI.DebugInstVec << SPIRVNL() << MI.FuncVec;
+                       MI.ForwardPointerVec);
+
+  if (M.isAllowedToUseExtension(ExtensionID::SPV_INTEL_inline_assembly)) {
+    O << SPIRVNL() << MI.AsmVec;
```

Comment:
If I understand correctly, `MI.AsmVec` contains only `OpAsmINTEL` instructions, which refer to `OpAsmTargetINTEL` ones. This is probably not a critical problem, but how the order is preserved? I.e. `OpAsmTargetINTEL` must go before `OpAsmINTEL` in order to be referenced, according to the SPIR-V spec

Reply:
Right now, the order is managed by OpAsmIntel.
OpAsmIntel calls encode for OpAsmTargetIntel before it encodes itself.